docs/trace/extending-the-sdk/README.md

```diff
@@ -211,6 +212,14 @@ Writing an instrumentation library typically involves 3 steps.
          > Changing the name of the source should be considered a
          breaking change.
 
+    > **Warning**
+    > If enabling the instrumentation requires references to objects
+    from the targeted library, your instrumentation library should
+    offer public APIs to receive them instead of relying on the extension methods
+    on `TracerProviderBuilder` and/or `IServiceCollection`, as the objects might
+    not be available yet at the time the tracer is being built
+    (for example, connections established later in the app lifetime).
```

Comment:
could you add an example/link to an instrumention which does this / require this? StackExchange.Redis?